build.gradle.kts

```
@@ -1,81 +1,5 @@
-plugins {
-	kotlin("jvm") version "1.9.25"
-	kotlin("plugin.spring") version "1.9.25"
-	kotlin("plugin.jpa") version "1.9.25"
-	kotlin("kapt") version "1.9.25"
-	id("org.springframework.boot") version "3.4.1"
-	id("io.spring.dependency-management") version "1.1.7"
-}
-
-group = "com.dobby"
-version = "0.0.1-SNAPSHOT"
-
-java {
-	toolchain {
-		languageVersion.set(JavaLanguageVersion.of(17))
-	}
-}
-
-tasks.getByName<Jar>("jar") {
-	enabled = true
-}
-
-
-repositories {
-	mavenCentral()
-}
-
-dependencies {
-	implementation("org.springframework.boot:spring-boot-starter-data-jpa")
-	implementation("org.springframework.boot:spring-boot-starter-validation")
-	implementation("org.springframework.boot:spring-boot-starter-mail")
-	implementation("org.springframework.boot:spring-boot-starter-web")
-	implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
-	implementation("org.springframework.boot:spring-boot-starter-actuator")
-	implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.3")
-	implementation("org.jetbrains.kotlinx:kotlinx-coroutines-reactor:1.7.3")
-	implementation("org.jetbrains.kotlin:kotlin-reflect")
-	implementation("org.springframework.boot:spring-boot-starter-security")
-	implementation("org.mariadb.jdbc:mariadb-java-client:2.7.3")
-	implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.7.0")
-	implementation("org.springframework.cloud:spring-cloud-starter-openfeign")
-	implementation("org.springframework.boot:spring-boot-starter-quartz")
-	implementation("org.springframework.boot:spring-boot-starter-thymeleaf")
-	implementation("io.awspring.cloud:spring-cloud-starter-aws:2.4.4")
-	implementation("software.amazon.awssdk:s3:2.20.59")
-	implementation("software.amazon.awssdk:ses:2.20.100")
-	implementation("org.springframework.boot:spring-boot-starter-data-redis")
-	implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.3")
-	implementation("org.springframework.boot:spring-boot-starter-aop")
-	implementation("ca.pjer:logback-awslogs-appender:1.6.0")
-
-	compileOnly("org.projectlombok:lombok")
-	runtimeOnly("com.mysql:mysql-connector-j")
-	runtimeOnly("com.h2database:h2")
-	annotationProcessor("org.projectlombok:lombok")
-}
-
-dependencyManagement {
-	imports {
-		mavenBom("org.springframework.cloud:spring-cloud-dependencies:2024.0.0")
-	}
-}
-
-dependencies {
-	implementation("org.slf4j:slf4j-api")
-	implementation("ch.qos.logback:logback-classic")
-}
-
-kotlin {
-	compilerOptions {
-		freeCompilerArgs.addAll("-Xjsr305=strict")
-	}
-}
-
-tasks.getByName<org.springframework.boot.gradle.tasks.bundling.BootJar>("bootJar") {
-	enabled = false
-}
-
-tasks.getByName<Jar>("jar") {
-	enabled = true
+subprojects {
+    repositories {
+        mavenCentral()
+    }
```

각 모듈에서 공통으로 사용할 부분은 subprojects로 관리하는 점 좋습니다!
아래처럼 `mavenCentral()`뿐만 아니라 `kotlin("jvm")`, `ktlint` 같은 플러그인도 함께 관리하면 더 깔끔할 것 같은데 어떻게 생각하시나요??

```suggestion
plugins {
    kotlin("jvm") version "1.9.25" apply false
    id("org.jlleitschuh.gradle.ktlint") apply false
}

subprojects {
    apply(plugin = "kotlin")
    apply(plugin = "org.jlleitschuh.gradle.ktlint")

    java {
        toolchain {
            languageVersion.set(JavaLanguageVersion.of(17))
        }
    }

    repositories {
        mavenCentral()
    }

    dependencies {
        implementation("org.jetbrains.kotlin:kotlin-stdlib")
    }
}
```

---

지수님 의견처럼 루트에서 공통 플러그인은 `apply(plugin = "...")` 로 일괄 적용하는 게
실제로 더 간편하고 실용적인 접근이라는 점에 완전히 공감합니다!

다만 이번 프로젝트는 클린 아키텍처 패턴을 strict하게 적용하는 게 목표이다 보니, 각 모듈에서 사용하는 플러그인을 명시적으로 선언해서 설정의 책임을 분리하고 모듈 간 의존성도 더 명확히 드러내는 방식에 저는 조금 더 마음이 끌리네요,,, 👀 😊
```kotlin
subprojects {
    apply(plugin = "org.jetbrains.kotlin.jvm")
    apply(plugin = "org.jlleitschuh.gradle.ktlint")

    kotlin {
        jvmToolchain {
            languageVersion.set(JavaLanguageVersion.of(17))
        }
    }

    repositories {
        mavenCentral()
    }

    dependencies {
        implementation("org.jetbrains.kotlin:kotlin-stdlib")
    }
}

```

혹시 요런 방식에 대해서는 어떻게 생각하시나요?

---

각 모듈에서 사용하는 플러그인과 라이브러리를 명확히 선언하는 차원에서, `group`과 `version`만 추가하는 것도 좋은 방법일 것 같아요! 👍